pkg/prometheus/server/operator.go

```diff
@@ -1742,3 +1791,66 @@ func (c *Operator) createOrUpdateWebConfigSecret(ctx context.Context, p *monitor
 
 	return nil
 }
+
+// checkCRDAuthorization checks if the operator has access to the specified CRD
+// It checks individual namespaces if an allowlist was provided, otherwise it checks cluster-wide.
+func (c *Operator) checkCRDAuthorization(ctx context.Context, verb string, group string, resource string) (bool, error) {
+	if c.config.Namespaces.AllowList != nil {
+		allowed := true
+		for ns := range c.config.Namespaces.AllowList {
+			ssar := &authv1.SelfSubjectAccessReview{
+				Spec: authv1.SelfSubjectAccessReviewSpec{
+					ResourceAttributes: &authv1.ResourceAttributes{
+						Verb:      verb,
+						Group:     group,
+						Resource:  resource,
+						Namespace: ns,
+					},
+				},
+			}
+			ssarResponse, err := c.kclient.AuthorizationV1().SelfSubjectAccessReviews().Create(ctx, ssar, metav1.CreateOptions{})
+			if err != nil {
+				return false, err
+			}
+			if !ssarResponse.Status.Allowed {
+				allowed = false
+				break
+			}
+		}
+
+		return allowed, nil
+	}
```

Comment:
Seems like we don't need `allowed`
```suggestion
	if c.config.Namespaces.AllowList != nil {
		for ns := range c.config.Namespaces.AllowList {
			ssar := &authv1.SelfSubjectAccessReview{
				Spec: authv1.SelfSubjectAccessReviewSpec{
					ResourceAttributes: &authv1.ResourceAttributes{
						Verb:      verb,
						Group:     group,
						Resource:  resource,
						Namespace: ns,
					},
				},
			}
			ssarResponse, err := c.kclient.AuthorizationV1().SelfSubjectAccessReviews().Create(ctx, ssar, metav1.CreateOptions{})
			if err != nil {
				return false, err
			}
			if !ssarResponse.Status.Allowed {
				return ssarResponse.Status.Allowed, nil
			}
		}

		return true, nil
	}
```

Reply:
Thanks! Fixed in fc6aa3ef7